rclpy/rclpy/duration.py

```diff
@@ -98,7 +108,7 @@ def from_msg(cls, msg):
             raise TypeError('Must pass a builtin_interfaces.msg.Duration object')
         return cls(seconds=msg.sec, nanoseconds=msg.nanosec)
 
-    def get_c_duration(self):
+    def get_c_duration(self) -> RCLDurationType:
```

Comment:
> The problem with that solution is without making some custom type alias mypy, and pylance have to treat _rclpy.rcl_duration_t as type Any which weakens the type inference. For example since the _duration_handle is type Any it cannot reason about the type of the nanoseconds field.
> 
> rclpy/rclpy/duration.py:44: error: Returning Any from function declared to return "int" [no-any-return]
> 
> ```python
> @property
>     def nanoseconds(self) -> int:
>         return self._duration_handle.nanoseconds
> ```

I see, thanks for the explanation. this does not break the userspace, right? (just checking) previously it returns `Any` but now `RCLDurationType`.

Reply:
Yes previously it has the type annotation `Any`. And would now have the type annotation `RCLDurationType`. The type annotation does not have any effective on what  the actual return type is which is  `_rclpy_pybind11.Duration` I believe.  Because type annotations are ignored at run time it has no effect on the actual code behavior. I'm not entirely clear what you mean by userspace.